arrow/arrow.py

```diff
@@ -1567,6 +1575,10 @@ def _get_iteration_params(cls, end, limit):
                 return end, sys.maxsize
             return end, limit
 
+    @staticmethod
+    def _is_last_day_of_month(date):
+        return (date + timedelta(days=1)).day == 1
```

Comment:
I am thinking it might be better to use the calendar module to handle this like in the following SO answer: https://stackoverflow.com/a/53591311/3820660. Adding a time delta might get dicey in the case of a DST transition. Using shift is also a possibility since that should account for imaginary datetimes (ref @systemcatch).

Reply:
Using `calendar` should help us avoid the dst and imaginary issues.